modules/contracts/adapters/dex/src/api.rs

```diff
@@ -132,6 +132,30 @@ impl<'a, T: DexInterface> Dex<'a, T> {
         })?;
         Ok(response)
     }
+
+    /// Generate the raw messages that are need to run a swap
+    pub fn generate_swap_messages(
+        &self,
+        offer_asset: OfferAsset,
+        ask_asset: AssetEntry,
+        max_spread: Option<Decimal>,
+        belief_price: Option<Decimal>,
+        sender_receiver: impl Into<String>,
+    ) -> AbstractSdkResult<SimulateSwapResponse> {
+        let response: SimulateSwapResponse = self.query(DexQueryMsg::GenerateMessages {
+            message: DexExecuteMsg::Action {
+                dex: self.dex_name(),
+                action: DexAction::Swap {
+                    offer_asset,
+                    ask_asset,
+                    max_spread,
+                    belief_price,
+                },
+            },
+            proxy_addr: sender_receiver.into(),
```

Comment:
This address is not restricted to the proxy, do we want to migrate this change from #214 here?

Reply:
Sure but that's the interface that is provided here. We will migrate that once the change is done. Here the api indicates `sender_receiver` which corresponds to that change